ledger-core/v2/virtual/execute/execute_test.go

```diff
@@ -156,14 +156,104 @@ func TestSMExecute_StartRequestProcessing(t *testing.T) {
 	assert.Equal(t, smObject.KnownRequests.Len(), 1)
 	assert.True(t, smObject.KnownRequests.GetList(contract.CallTolerable).Exist(smExecute.execution.Outgoing))
 
-	{ // update known requests panics
+	{
+		// call same request
+		// expect duplicate detection and SM stop
+		wasStoped := false
+
 		execCtx := smachine.NewExecutionContextMock(mc).
-			UseSharedMock.Set(shareddata.CallSharedDataAccessor)
+			UseSharedMock.Set(shareddata.CallSharedDataAccessor).
+			LogMock.Return(smachine.Logger{}).
```

Comment:
We need some way to create "testing" instance of `smachine.Logger{}` here, that will use "context" of background

Reply:
Maybe, but it's unnecessary in this test case

Reply:
@bigbes

Reply:
I suggest to use /dev/null Logger here, but it's not important